docs/manual/howto/widget.rst

```diff
@@ -433,6 +433,63 @@ For example, the Mpris2 widget uses the following code:
 interfaces. Refer to the `dbus-next documentation <https://python-dbus-next.readthedocs.io/en/latest/>`_
 for more information on how to use the module.
 
+Mouse events
+============
+
+By default, widgets handle button presses and will call any function that is bound to the button in the
+``mouse_callbacks`` dictionary. The dictionary keys are as follows:
+
+ - ``Button1``: Left click
+ - ``Button2``: Middle click
+ - ``Button3``: Right click
+ - ``Button4``: Scroll up
+ - ``Button5``: Scroll down
+ - ``Button6``: Scroll left
+ - ``Button7``: Scroll right
+
+You can then define your button bindings in your widget (e.g. in ``__init__``):
+
+.. code:: python
+
+    class MyWidget(widget.TextBox)
+
+        def __init__(self, *args, **config):
+            widget.TextBox.__init__(self, *args, **kwargs)
+            self.add_callbacks(
+                {
+                    "Button1": self.left_click_method,
+                    "Button3": self.right_click_method
```

Comment:
Lazy objects can't be used here, can they? I remember once this was asked and there have been changes to them so wondering if that is now possible. Reading the code it doesn't look like they could be - calling a lazy object is only useful for providing arguments, but not executing the lazy call. Is that right?

Reply:
Lazy calls can be added. This is different to the changes to lazy calls to pass args.

I'll add a bit to show that lazy calls cab be bound to buttons too.

Reply:
Ah, the handling code is in `widget.base`, gotcha. Neat.